File: backend/src/schemas/project.py
"""Project Pydantic schemas."""

from datetime import datetime
from src.schemas import AppBaseSchema


class ProjectCreate(AppBaseSchema):
    name: str
    creative_description: str
    template_id: int
    working_dir_root: str = "/Users/wangchangbin/ai/vimax_ru"


class ProjectConfigUpdate(AppBaseSchema):
    yaml_content: str
    config_py_content: str


class StepSummary(AppBaseSchema):
    total: int
    completed: int
    failed: int


class ProjectListItem(AppBaseSchema):
    id: int
    name: str
    creative_description: str
    working_dir: str
    status: str
    template_name: str | None = None
    current_step_name: str | None = None
    step_summary: StepSummary | None = None
    created_at: datetime
    updated_at: datetime


class ProjectConfig(AppBaseSchema):
    yaml_content: str
    config_py_content: str


class ProjectResponse(AppBaseSchema):
    id: int
    name: str
    creative_description: str
    working_dir: str
    status: str
    template_id: int | None = None
    current_step_name: str | None = None
    config: ProjectConfig | None = None
    unconfirmed_count: int = 0
    created_at: datetime
    updated_at: datetime


class ProjectListResponse(AppBaseSchema):
    projects: list[ProjectListItem]
    total: int
    page: int
    page_size: int


class ProjectCreateResponse(AppBaseSchema):
    id: int
    name: str
    working_dir: str
    status: str
    created_at: datetime


class ProgressStepSchema(AppBaseSchema):
    """A single progress step (from template's progress.py)."""
    name: str          # internal key, e.g. "story"
    label: str         # human-readable, e.g. "生成故事"
    order: int         # zero-based index
    status: str        # "new" | "running" | "success" | "failed" | "pending"


class ProjectProgressResponse(AppBaseSchema):
    """Full progress enum for a project."""
    steps: list[ProgressStepSchema]
    current_step_order: int | None = None  # currently executing step index

File: frontend/src/types/project.ts
export interface ProjectListItem {
  id: number;
  name: string;
  creative_description: string;
  working_dir: string;
  status: "idle" | "running" | "completed" | "failed";
  template_name: string;
  current_step_name: string | null;
  step_summary: { total: number; completed: number; failed: number };
  created_at: string;
  updated_at: string;
}

export interface ProjectDetail {
  id: number;
  name: string;
  creative_description: string;
  working_dir: string;
  status: "idle" | "running" | "completed" | "failed";
  template_id: number;
  current_step_name: string | null;
  config: {
    yaml_content: string;
    config_py_content: string;
  };
  unconfirmed_count: number;
  created_at: string;
  updated_at: string;
}

export interface ProjectCreateRequest {
  name: string;
  creative_description: string;
  template_id: number;
  working_dir_root: string;
}

export interface ProjectConfigUpdateRequest {
  yaml_content: string;
  config_py_content: string;
}

export interface ProjectListResponse {
  projects: ProjectListItem[];
  total: number;
  page: number;
  page_size: number;
}

export interface ProgressStep {
  name: string;
  label: string;
  order: number;
  status: "new" | "running" | "success" | "failed" | "pending";
}

export interface ProjectProgressResponse {
  steps: ProgressStep[];
  current_step_order: number | null;
}
